packages/cloud_firestore/lib/src/firestore.dart

```diff
@@ -52,4 +60,64 @@ class Firestore {
     assert(path != null);
     return new DocumentReference._(this, path.split('/'));
   }
+
+  /// Runs a set of atomic database operations.
+  Future<Map<String, dynamic>> runTransaction(
+      TransactionHandler transactionHandler,
+      {Duration timeout: const Duration(seconds: 5)}) async {
+    assert(timeout.inMilliseconds > 0,
+        'Transaction timeout must be more than 0 milliseconds');
+    final int transactionId = _transactionHandlerId++;
+    _transactionHandlers[transactionId] = transactionHandler;
+    final Map<String, dynamic> result = await channel.invokeMethod(
+        'Firestore#runTransaction', <String, dynamic>{
+      'transactionId': transactionId,
+      'transactionTimeout': timeout.inMilliseconds
+    });
+    return result ?? <String, dynamic>{};
+  }
+}
+
+typedef Future<Map<String, dynamic>> TransactionHandler(Transaction tx);
+
+class Transaction {
+  MethodChannel _channel;
+  int _transactionId;
+
+  Transaction(this._transactionId) : _channel = Firestore.channel;
+
+  Future<DocumentSnapshot> get(DocumentReference documentReference) async {
+    final dynamic result = await _channel.invokeMethod(
+        'Transaction#get', <String, dynamic>{
+      'transactionId': _transactionId,
+      'path': documentReference.path
+    });
+    return new DocumentSnapshot._(
```

Comment:
We should probably `null` check result here and return a `null` instead of a `DocumentSnapshot` for the case where it doesn't exist.

Reply:
Done!